### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Rian501**
Updates:

- README documentation, especially on tools, build, and deploy
- user passwords can now use symbols
- email-sending function firing properly
- prevents default behavior of form submission allowing controlled functionality

### CODE REVIEW

**Comment by Rian501** [comment]
The dev branch is what is currently deployed.

**Comment by CoreyAR** [comment]
I think this PR is looking good. We should update the description with what it contains. We can add notes about  the changes to the README, the strings, and dependencies/ package.json

File: README.md [hunk 1]
```diff
@@ -1,39 +1,73 @@
 # Glean Tennessee
-Preview [website](https://gleantn-1794b.firebaseapp.com/#)
 
-This project was created during Nashville's Civic Day of Hacking to allow farmers and other food providers to notify the [Society of St. Andrew](endhunger.org) of food donations available for harvest and pick up.
+Preview [website]
```
Comment (combinatorist) [inline README.md]: Update website (to SOSA domain) and use markdown links at the end of doc.

File: README.md [hunk 2]
```diff
@@ -1,39 +1,73 @@
...
 This project was bootstrapped with [Create React App](https://github.com/facebookincubator/create-react-app).
 
-To start:
+To start the app locally, do:
 
 ```
-npm install
-npm start
+yarn install
+yarn start
```
Comment (combinatorist) [inline README.md]: Switch to yarn by default (with caveat below).

File: README.md [hunk 3]
```diff
@@ -1,39 +1,73 @@
...
+
+```
+yarn build
+firebase deploy
+```
+
+This is only necessary for changes that affect the app (the UI). For changes to background functions, see below.
+
+> If you get warnings about firebase as a peer dependency when you do `yarn install` then you will need to manually `yarn add firebase-admin` before you deploy.
+> If you get a warning about firebase functions, do an `npm install` in the functions directory as suggested and then re-run `yarn build` before you try again with `firebase deploy`.
```
Comment (combinatorist) [inline README.md]: Describe Deployment including firebase.

File: README.md [hunk 4]
```diff
@@ -1,39 +1,73 @@
...
+`$ firebase deploy --only functions`
 
 The function code lives in the `functions` dir.
 
+## Testing
+
+You can test the code by starting the app locally with npm or yarn (above).
+
+Our target is for the app to be displayed on the Society of St. Andrew [website](https://endhunger.org/crops.html).
+But in case something is wrong on their site, you try our [firebase deployment](https://glean-tennessee.firebaseapp.com).
```
Comment (combinatorist) [inline README.md]: Add testing section

File: README.md [hunk 5]
```diff
@@ -1,39 +1,73 @@
...
+* Jeannie Hunter
+* Emily Lemmon
+* Kevin Huber
+* Shu Sajid
+* Andrew Leverette
+
+[code for nashville]: http://www.codefornashville.org/
+[firebase deployment]: https://glean-tennessee.firebaseapp.com/
+[website]: https://endhunger.org/crops.html
+[society of st. andrew]: https://endhunger.org
```
Comment (combinatorist) [inline README.md]: Some Markdown style (different bullets may have come from a linter) and I made these key value links to keep text less cluttered.

File: functions/package.json [hunk 6]
```diff
@@ -10,8 +10,11 @@
   },
   "dependencies": {
     "@sendgrid/mail": "^6.3.1",
-    "firebase-admin": "~6.0.0",
-    "firebase-functions": "^2.1.0"
+    "firebase-admin": "^8.2.0",
+    "firebase-functions": "^3.0.1"
```
Comment (combinatorist) [inline functions/package.json]: firebase update may have been a consequence of going to node 8.

File: functions/package.json [hunk 7]
```diff
@@ -10,8 +10,11 @@
...
   },
-  "private": true
+  "private": true,
+  "engines": {
+    "node": "8"
```
Comment (combinatorist) [inline functions/package.json]: Don't know what version of node was used in practice before, but got it working with 8 and made it explicit to help the next person.

File: package.json [hunk 8]
```diff
@@ -12,8 +12,9 @@
     "lint:fix": "prettier --write 'src/**/*.{js,json,jsx}'"
   },
   "dependencies": {
+    "ajv": "^6.10.0",
```
Comment (combinatorist) [inline package.json]: Not sure whether updates were necessary in both top level and functions directory `package.json` - perhaps from doing `firebase deploy ---only functions`?

File: src/components/profile-form/index.js [hunk 9]
```diff
@@ -12,7 +12,8 @@ const Validators = {
   email: {regex: Regex.email, message: 'Please enter a valid email.'},
   password: {
     regex: Regex.password,
-    message: 'Please enter a password at least 8 characters long.'
+    message:
+      'Please enter a password at least 8 characters long which contains only letters (upper or lowercase) and numbers.'
```
Comment (combinatorist) [inline src/components/profile-form/index.js]: Lol - this was the original impetus for everything else (figuring out and documenting how to install, build, and deploy).

File: src/screens/login/index.js [hunk 10]
```diff
@@ -108,7 +108,7 @@ export default class Login extends Component {
             <p>
               Our volunteers will arrive to glean your produce, and we will
               deliver the food to food banks, churches, pantries, and other
-              agencies at no cost to you.
+              agencies.
```
Comment (combinatorist) [inline src/screens/login/index.js]: This change might predate me.

**Comment by enlore** [comment]
@Rian501 please add notes to the description for the PR describing what all is in here (or at least a high level summary).

**Pull request merged**